designdocs/provider_registration.md

```diff
@@ -0,0 +1,185 @@
+# Provider Registration
+
+## Table of Contents
+* [Context and Scope](#context-and-scope)
+* [Problem Statement](#problem-statement)
+* [Goals and Non-Goals](#goals-and-non-goals)
+* [Proposal](#proposal)
+* [Benefits and Risks](#benefits-and-risks)
+* [Alternatives and Rationale](#alternatives-and-rationale)
+* [Glossary](#glossary)
+
+## Context and Scope
+In order to properly implement Frequency, we need to describe how service providers
+will participate on the network.
+
+## Problem Statement
+Service providers are an integral part of how Frequency functions. However, there are
+some open questions about how they operate, questions that this document hopes
+to provide clarity on.
+
+Among those questions are:
+
+1. Who can become a provider?
+1. How does a user become a provider?
+1. What are the attributes of a provider?
+1. How will providers be identified to other network actors?
+1. How does one stop being a provider?
+1. Can provider status be revoked from a provider?
+
+## Goals and Non-Goals
+This document will discuss the ways in which service providers can register with
+and participate in the Frequency network.
+
+This document will not discuss the specifics of how providers will process
+service requests from users on the network. This document will also not discuss
+specifics around creating and honoring transactions between users.
+
+## Proposal
+The basic workflow for provider registration is as follows:
+1. User stakes balance to become a provider. This is called a "Registration Fee".
+1. The "Registration Fee" is credited to the Chain Treasury (TBD).
+1. The provider's MSA id is added to storage on chain.
```

Comment:
Not exactly connected to this line, but close...

I don't understand the need to have the Announcement structures or quite what the announcement structures do.

- `ProviderAnnouncementMetadata` -> `ProviderMetadata`
- `ProviderRegistrationAnnouncement` -> Event, not storage
- `ProviderUnregistrationAnnouncement` -> Event, not storage